main.cpp

```diff
@@ -121,196 +121,226 @@ struct CarWash
  4) After you finish defining each type/function, click the [run] button.  Clear up any errors or warnings as best you can.
  */
 
+//--------------------------------------------------------
+// DrillMachine
+//--------------------------------------------------------
 
-
-/*
-Thing 1) Drill Machine
-5 properties:
-    1) size of drill chucks
-    2) number of modes
-    3) motor power
-    4) cable length 
-    5) drilling speed
-3 things it can do:
-    1) drill hole
-    2) screw screw
-    3) load the accu
- */
-
-struct DrillMaschine 
+struct DrillMachine 
 {
-    // Size of the drill chucks
     int sizeOfDrillChucks = 10;
-    // Number of modes
     int numOfModes = 2;
-    // Motor power 
     int motorPower = 1500;
-    // Cable length
     float cableLength = 1.5f;
-    // Drilling speed
-    int drillingSpeed = 3000;
+    bool accuTooWeak = false;
+    int screwTime = 0;
 
     struct DrillBit
     {
-        float drillChuck = 10.0f;
+        int drillChuck = 10;
         int length = 140;
         std::string material = "steel";
         bool isHardened = true;
         bool isForWood = true;
+        bool isStucked = false;
+        float maxTemperature = 303.3f;
 
         void drill(int depth = 10);
-        bool isStuck();
-        bool isToHot();
-
-
+        bool isDull(float currentTemperature);
+        bool isToHot(float currentTemperature);
     };
 
-    // Drill hole (return true if screw process was sucessful)
     bool drillHole(DrillBit bit);
-    // screw screws (return true if screw process was sucessful)
     bool screwScrew(int speed, bool screwIn);
-    // load the accu
     void loadAccu();
 
     DrillBit myDrillBit;
 };
 
 
-/*
-Thing 2) Soda Maker
-5 properties:
-    1) water pressure
-    2) size of container
-    3) volume of soda tank 
-    4) years of guarantee
-    5) number of bottles
-3 things it can do:
-    1) make soda water
-    2) install soda tank
-    3) get name of the manufacturer
- */
+void DrillMachine::DrillBit::drill(int depth) {
+    if(depth > length) isStucked = true;
+}
+
+
+bool DrillMachine::DrillBit::isDull(float currentTemperature) {
+    if(isToHot(currentTemperature)) return true;
+    else return false;
+}
```

Comment:
curly braces go on their own line
If you find yourself writing code like this:
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    else 
    {
        return anotherValue;
    }
}
```
if the `if()` block is entered, the `return aValue;` is the last line of code to be executed.
you have no code after the `else` block that needs executing, since your `else` block contains a `return` statement.
`return` statements are the last thing that **ever happens** in a function
That means if your `if()` block is **NOT** entered, it will be skipped.  What is the next thing that happens?  `return anotherValue;`
that means you do **NOT** need the else block, and it is just causing visual clutter.
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    return anotherValue;
}
```

you can omit the `if/else` entirely by simply writing:
```
return isToHot(currentTemperature);
```

- [ ] revise